src/agoradatatools/process.py

```diff
@@ -123,12 +125,22 @@ def process_dataset(
             staging_path=staging_path,
             filename=dataset_name + "." + dataset_obj[dataset_name]["final_format"],
         )
-    else:
+    elif isinstance(df, list):
+        json_path = load.list_to_json(
+            df=df,
+            staging_path=staging_path,
+            filename=dataset_name + "." + dataset_obj[dataset_name]["final_format"],
+        )
+    elif isinstance(df, DataFrame):
```

Comment:
@BWMac here are the changes mentioned in the PR description. What do you think about them?

Reply:
I also felt the variable `df` was a bit confusing, I'll change the variable name.

Reply:
Actually, I took a look and it seems that the variable `df` is being used a lot. Maybe it would be best I not touch it in this PR.

Feedback request: I kept the output data as a `list` because I felt it was most efficient. But if the priority is maintainability, I could transform it into a `pd.DataFrame` or a `dict`. Neither option is too much work and I would prefer to do what is best for the team, what do you think is best in this case?

Reply:
I think keeping it as a list is fine. You may want to add a `Union` type hint [here](https://github.com/Sage-Bionetworks/agora-data-tools/blob/d2b1ba03c5397250d2113e210c4947710c474cee/src/agoradatatools/process.py#L20) so that it is clear that the output of a transform can be any of these three data types.